Support for user block from the Adapter

redaguermas opened the pull request:
<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
We currently do not have a good way to block user traffic for sites that are malicious. This change will allow us to reduce calls from users that we don't want to have traffic from. 

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- reda.guermas@nobid.io
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

redaguermas commented:
Hello,
Is this good to go?

redaguermas commented:
@robertrmartinez ,
Can you please review and let me if you can merge it please

mkendall07 commented on modules/nobidBidAdapter.js:
```diff
@@ -11,6 +11,27 @@ window.nobid.refreshCount = 0;
 function log(msg, obj) {
   utils.logInfo('-NoBid- ' + msg, obj)
 }
+function nobidSetCookie(cname, cvalue, hours) {
```
please use `utils.setCookie()` instead of doing it in the adapter.

mkendall07 commented on modules/nobidBidAdapter.js:
```diff
@@ -11,6 +11,27 @@ window.nobid.refreshCount = 0;
+  var ccval = cname + "=" + cvalue + ";" + expires + ";path=/";
+  document.cookie = ccval;
+}
+function nobidGetCookie(cname) {
```
use `utils.getCookie()`

robertrmartinez commented:
Agree with @mkendall07 

Also please add unitTests where possible.

The noBid adapter has dropped below the 80% threshold for test coverage.

![image](https://user-images.githubusercontent.com/13972794/73397253-f3713900-4297-11ea-8aed-f57a4fc9e41d.png)

redaguermas commented:
@robertrmartinez ,
How did you get the test coverage? I am running "gulp test --coverage"  but I am unable to find the report.
@mkendall07 ,
Thanks for the feedback. Will fix.

GLStephen commented:
@redaguermas the test coverage is put in the builds folder, there is an index.html you can use to view it.

robertrmartinez commented:
Thank you @GLStephen for the quick response.

Yes in the build folder after the run completes you will see a `coverage` folder as well.

I just start up a simple web server in the build dir to see it.

![image](https://user-images.githubusercontent.com/13972794/73481592-009e2e80-4351-11ea-9607-a5f3cc69aa61.png)

Karma HTML will show you results for all modules and tested files.

redaguermas commented:
I am not sure what the error in the CI is?
It's related to browser not being able to load.

redaguermas commented:
@robertrmartinez , I added more unit tests.
@mkendall07 , I changed get and set cookies to use the utils class.
The CI is broken. I think it's independent of my changes.

robertrmartinez commented:
It looks like the Circle Ci tests are running on your own circle CI instance:

![image](https://user-images.githubusercontent.com/13972794/73886432-748f7980-481e-11ea-879a-0bb073fa3fbf.png)

The workaround is the following to get it to run on the Prebid Version:

```
On your CircleCI project's screen, click "unfollow" on the Prebid.js row for your fork:
```
![image](https://user-images.githubusercontent.com/13972794/73886502-a0126400-481e-11ea-92f0-abbff498e885.png)

If you can please do this so we can get the Prebid tests running successfully.

redaguermas commented:
@robertrmartinez , @mkendall07 , @GLStephen 
Can you please review this release. We need it out ASAP for our Publishers.
Thanks

robertrmartinez commented:
@redaguermas Please see my most recent comment.

Need you to unfollow your own repo on circle CI so we can run them on the Prebid version of Circler CI

Thanks.

robertrmartinez commented:
@redaguermas 

You have two lint errors in the noBidBidAdapter.js

```
/home/circleci/Prebid.js/modules/nobidBidAdapter.js
  331:1  error  Expected indentation of 8 spaces but found 10  indent
  332:1  error  Expected indentation of 8 spaces but found 10  indent
```

Please fix them so Circle CI Lint passes.

redaguermas commented:
Great. Passed CI. Thanks @robertrmartinez  for your help.

redaguermas commented:
@robertrmartinez , @GLStephen , @mkendall07 
All changes have been done and the tests passed. Can you please review?
